CMake build error when importing lzma on darwin

# Bug report

### What operating system and version are you using?

Latest master branch.
macOS arm64 14.5 (23F79)

### What version of osquery are you using?

None, trying to build from master.

### What steps did you take to reproduce the issue?

Follow build instructions for a fresh osquery repo clone
https://osquery.readthedocs.io/en/stable/development/building/#step-2-download-and-build-source-on-macos

At the configure step `cmake -DCMAKE_OSX_DEPLOYMENT_TARGET=10.15 -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ ..`, receiving an error when importing source/lzma as it runs into an error cloning `https://git.tukaani.org/xz.git/`.

```
# Configure
mkdir build; cd build
cmake -DCMAKE_OSX_DEPLOYMENT_TARGET=10.15 -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ ..
```

I tried different networks, deleted the build folder, etc. and receive the same 404 on the site https://git.tukaani.org/xz.git/. 

### What did you expect to see?

Successful configuration of osquery locally.

### What did you see instead?

```
Failed to clone 'libraries/cmake/source/lzma/src'. Retry scheduled
07:53:53.727730 run-command.c:657       trace: run_command: git submodule--helper clone --prefix libraries/cmake/source/lzma/ --path libraries/cmake/source/lzma/src --name libraries/cmake/source/lzma/src --url https://git.tukaani.org/xz.git
07:53:53.876146 git.c:465               trace: built-in: git submodule--helper clone --prefix libraries/cmake/source/lzma/ --path libraries/cmake/source/lzma/src --name libraries/cmake/source/lzma/src --url https://git.tukaani.org/xz.git
07:53:53.876851 run-command.c:657       trace: run_command: unset GIT_PREFIX; GIT_DIR=.git git clone --no-checkout --separate-git-dir /Users/antmartinez/go/src/osquery/.git/modules/libraries/cmake/source/lzma/src -- https://git.tukaani.org/xz.git /Users/antmartinez/go/src/osquery/libraries/cmake/source/lzma/src
07:53:53.886878 git.c:465               trace: built-in: git clone --no-checkout --separate-git-dir /Users/antmartinez/go/src/osquery/.git/modules/libraries/cmake/source/lzma/src -- https://git.tukaani.org/xz.git /Users/antmartinez/go/src/osquery/libraries/cmake/source/lzma/src
Cloning into '/Users/antmartinez/go/src/osquery/libraries/cmake/source/lzma/src'...
07:53:53.969176 run-command.c:657       trace: run_command: git remote-https origin https://git.tukaani.org/xz.git
07:53:54.110884 git.c:750               trace: exec: git-remote-https origin https://git.tukaani.org/xz.git
07:53:54.112917 run-command.c:657       trace: run_command: git-remote-https origin https://git.tukaani.org/xz.git
07:53:54.132752 http.c:843              == Info: Couldn't find host git.tukaani.org in the .netrc file; using defaults
07:53:54.136378 http.c:843              == Info: Host git.tukaani.org:443 was resolved.
07:53:54.136385 http.c:843              == Info: IPv6: (none)
07:53:54.136387 http.c:843              == Info: IPv4: 5.44.245.25
07:53:54.136410 http.c:843              == Info:   Trying 5.44.245.25:443...
07:53:54.290755 http.c:843              == Info: Connected to git.tukaani.org (5.44.245.25) port 443
07:53:54.290819 http.c:843              == Info: ALPN: curl offers h2,http/1.1
07:53:54.290945 http.c:843              == Info: (304) (OUT), TLS handshake, Client hello (1):
07:53:54.294443 http.c:843              == Info:  CAfile: /etc/ssl/cert.pem
07:53:54.294453 http.c:843              == Info:  CApath: none
07:53:54.484943 http.c:843              == Info: (304) (IN), TLS handshake, Server hello (2):
07:53:54.485085 http.c:843              == Info: (304) (IN), TLS handshake, Unknown (8):
07:53:54.485576 http.c:843              == Info: (304) (IN), TLS handshake, Certificate (11):
07:53:54.486165 http.c:843              == Info: (304) (IN), TLS handshake, CERT verify (15):
07:53:54.486276 http.c:843              == Info: (304) (IN), TLS handshake, Finished (20):
07:53:54.486358 http.c:843              == Info: (304) (OUT), TLS handshake, Finished (20):
07:53:54.486371 http.c:843              == Info: SSL connection using TLSv1.3 / AEAD-AES256-GCM-SHA384 / [blank] / UNDEF
07:53:54.486374 http.c:843              == Info: ALPN: server accepted h2
07:53:54.486378 http.c:843              == Info: Server certificate:
07:53:54.486382 http.c:843              == Info:  subject: CN=git.tukaani.org
07:53:54.486387 http.c:843              == Info:  start date: Jun  9 20:33:58 2024 GMT
07:53:54.486389 http.c:843              == Info:  expire date: Sep  7 20:33:57 2024 GMT
07:53:54.486394 http.c:843              == Info:  subjectAltName: host "git.tukaani.org" matched cert's "git.tukaani.org"
07:53:54.486399 http.c:843              == Info:  issuer: C=US; O=Let's Encrypt; CN=R10
07:53:54.486403 http.c:843              == Info:  SSL certificate verify ok.
07:53:54.486435 http.c:843              == Info: using HTTP/2
07:53:54.486460 http.c:843              == Info: [HTTP/2] [1] OPENED stream for https://git.tukaani.org/xz.git/info/refs?service=git-upload-pack
07:53:54.486463 http.c:843              == Info: [HTTP/2] [1] [:method: GET]
07:53:54.486464 http.c:843              == Info: [HTTP/2] [1] [:scheme: https]
07:53:54.486466 http.c:843              == Info: [HTTP/2] [1] [:authority: git.tukaani.org]
07:53:54.486468 http.c:843              == Info: [HTTP/2] [1] [:path: /xz.git/info/refs?service=git-upload-pack]
07:53:54.486470 http.c:843              == Info: [HTTP/2] [1] [user-agent: git/2.45.2]
07:53:54.486472 http.c:843              == Info: [HTTP/2] [1] [accept: */*]
07:53:54.486474 http.c:843              == Info: [HTTP/2] [1] [accept-encoding: deflate, gzip]
07:53:54.486477 http.c:843              == Info: [HTTP/2] [1] [accept-language: en-US, *;q=0.9]
07:53:54.486479 http.c:843              == Info: [HTTP/2] [1] [pragma: no-cache]
07:53:54.486481 http.c:843              == Info: [HTTP/2] [1] [git-protocol: version=2]
07:53:54.486574 http.c:790              => Send header, 0000000224 bytes (0x000000e0)
07:53:54.486581 http.c:802              => Send header: GET /xz.git/info/refs?service=git-upload-pack HTTP/2
07:53:54.486583 http.c:802              => Send header: Host: git.tukaani.org
07:53:54.486584 http.c:802              => Send header: User-Agent: git/2.45.2
07:53:54.486607 http.c:802              => Send header: Accept: */*
07:53:54.486610 http.c:802              => Send header: Accept-Encoding: deflate, gzip
07:53:54.486611 http.c:802              => Send header: Accept-Language: en-US, *;q=0.9
07:53:54.486613 http.c:802              => Send header: Pragma: no-cache
07:53:54.486614 http.c:802              => Send header: Git-Protocol: version=2
07:53:54.486616 http.c:802              => Send header:
07:53:54.731202 http.c:790              <= Recv header, 0000000013 bytes (0x0000000d)
07:53:54.731215 http.c:802              <= Recv header: HTTP/2 404
07:53:54.731219 http.c:790              <= Recv header, 0000000015 bytes (0x0000000f)
07:53:54.731221 http.c:802              <= Recv header: server: nginx
07:53:54.731223 http.c:790              <= Recv header, 0000000037 bytes (0x00000025)
07:53:54.731224 http.c:802              <= Recv header: date: Wed, 24 Jul 2024 14:53:54 GMT
07:53:54.731226 http.c:790              <= Recv header, 0000000019 bytes (0x00000013)
07:53:54.731228 http.c:802              <= Recv header: content-length: 0
07:53:54.731229 http.c:790              <= Recv header, 0000000040 bytes (0x00000028)
07:53:54.731231 http.c:802              <= Recv header: expires: Fri, 01 Jan 1980 00:00:00 GMT
07:53:54.731232 http.c:790              <= Recv header, 0000000018 bytes (0x00000012)
07:53:54.731233 http.c:802              <= Recv header: pragma: no-cache
07:53:54.731235 http.c:790              <= Recv header, 0000000053 bytes (0x00000035)
07:53:54.731236 http.c:802              <= Recv header: cache-control: no-cache, max-age=0, must-revalidate
07:53:54.731237 http.c:790              <= Recv header, 0000000044 bytes (0x0000002c)
07:53:54.731239 http.c:802              <= Recv header: strict-transport-security: max-age=5184000
07:53:54.731240 http.c:790              <= Recv header, 0000000018 bytes (0x00000012)
07:53:54.731245 http.c:802              <= Recv header: vary: User-Agent
07:53:54.731247 http.c:790              <= Recv header, 0000000002 bytes (0x00000002)
07:53:54.731248 http.c:802              <= Recv header:
07:53:54.731259 http.c:843              == Info: Connection #0 to host git.tukaani.org left intact
fatal: repository 'https://git.tukaani.org/xz.git/' not found
fatal: clone of 'https://git.tukaani.org/xz.git' into submodule path '/Users/antmartinez/go/src/osquery/libraries/cmake/source/lzma/src' failed
Failed to clone 'libraries/cmake/source/lzma/src' a second time, aborting
07:53:54.739283 run-command.c:1549      run_processes_parallel: done
CMake Error at libraries/cmake/source/modules/utils.cmake:55 (message):
  Failed to update the following git submodule:
  "/Users/antmartinez/go/src/osquery/libraries/cmake/source/lzma/src"
Call Stack (most recent call first):
  libraries/cmake/source/modules/utils.cmake:149 (initializeGitSubmodule)
  libraries/cmake/source/modules/Findlzma.cmake:10 (importSourceSubmodule)
  CMakeLists.txt:168 (find_package)
  CMakeLists.txt:57 (importLibraries)
  CMakeLists.txt:218 (main)


-- Configuring incomplete, errors occurred!
```

--- a/.gitmodules
+++ b/.gitmodules
@@ -15,7 +15,7 @@
 	url = https://github.com/arangodb/linenoise-ng
 [submodule "libraries/cmake/source/lzma/src"]
 	path = libraries/cmake/source/lzma/src
-	url = https://git.tukaani.org/xz.git
+	url = https://github.com/tukaani-project/xz
 [submodule "libraries/cmake/source/rapidjson/src"]
 	path = libraries/cmake/source/rapidjson/src
 	url = https://github.com/Tencent/rapidjson

--- a/libraries/cmake/formula/openssl/CMakeLists.txt
+++ b/libraries/cmake/formula/openssl/CMakeLists.txt
@@ -187,12 +187,9 @@ function(opensslMain)
   list(APPEND openssl_c_flags ${OSQUERY_FORMULA_CFLAGS})
   string(REPLACE ";" " " openssl_c_flags "${openssl_c_flags}")
 
-  string(REGEX MATCH "[0-9]\.[0-9]\.[0-9]" OPENSSL_VERSION_NO_PATCH "${OPENSSL_VERSION}")
-
   if("${OSQUERY_OPENSSL_ARCHIVE_PATH}" STREQUAL "")
     set(openssl_urls
-      "https://www.openssl.org/source/openssl-${OPENSSL_VERSION}.tar.gz"
-      "https://www.openssl.org/source/old/${OPENSSL_VERSION_NO_PATCH}/openssl-${OPENSSL_VERSION}.tar.gz"
+      "https://github.com/openssl/openssl/releases/download/openssl-${OPENSSL_VERSION}/openssl-${OPENSSL_VERSION}.tar.gz"
     )
   else()
     if(NOT EXISTS "${OSQUERY_OPENSSL_ARCHIVE_PATH}" OR IS_DIRECTORY "${OSQUERY_OPENSSL_ARCHIVE_PATH}")

